src/components/container/SurveyDataChart/SurveyDataChart.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useState, useContext } from 'react'
+import { DateRangeContext } from '../../presentational/DateRangeCoordinator/DateRangeCoordinator'
+import { add, parseISO } from 'date-fns'
+import { SurveyAnswer } from '@careevolution/mydatahelps-js'
+import { WeekStartsOn, getMonthStart, getWeekStart } from '../../../helpers/get-interval-start'
+import TimeSeriesChart, { AreaChartOptions, BarChartOptions, LineChartOptions } from '../../presentational/TimeSeriesChart/TimeSeriesChart'
+import queryAllSurveyAnswers from '../../../helpers/query-all-survey-answers'
+import format from 'date-fns/format'
+import { useInitializeView } from '../../../helpers/Initialization'
+
+export interface SurveyAnswerChartParameters {
+    label: string;
+    surveyName?: string | string[];
+    stepIdentifier?: string | string[];
+    resultIdentifier: string | string[];
+}
+
+export interface SurveyDataChartProps {
+    title?: string
+    intervalType?: "Week" | "Month" | "SixMonth"
+    weekStartsOn?: WeekStartsOn
+    charts: SurveyAnswerChartParameters[],
+    chartType: "Line" | "Bar" | "Area"
+    options?: LineChartOptions | BarChartOptions | AreaChartOptions,
+    expectedDataInterval?: Duration,
```

Comment:
Sure - the FFWEL example that I keep going back to is expected at a monthly cadence.  So you could put an `expectedDataInterval` of `{months: 1}` here, and missing a month would come up with a gap in the chart.  Similarly that same program has a survey that's a Weekly Goal Check In where participants give a weekly update on a scale of 1-10 on how they're keeping up with their goals.  In that case the graph could be set to `expectedDataInterval: {weeks: 1}`.  In my customized version of that graph they actually wanted to graph "missed" weeks with an X icon which I think would be a pretty cool addition to this graph as an option, but I will keep that for a second pass.

So, the behavior is just like the DailyDataChart, except the user can set the expected cadence.

![image](https://github.com/CareEvolution/MyDataHelpsUI/assets/5473619/e09bc8e2-6ead-4fd3-812e-f5879cfd8b05)

Reply:
Ah...I'm sorry, I completely misunderstood/conflated what this was doing - probably because the storybook has an exact cadence with the graph - data is generated exactly 1 month apart.  So I assumed we actually were *grouping the data* by month; but we are not; data is still grouped by day for survey answers regardless, but we do not bother to "connect" days.  Your comment about how this is helping to display "Gaps" in the data also makes more sense now.

So yeah this makes sense...for daily data charts and such the expected cadence can always be hardcoded...for something like this I see where you could have an irregular cadence and want to specify over what time periods you have a "gap" in the graph...k cool